src/SignalR/clients/ts/signalr/tests/HttpConnection.test.ts

```diff
@@ -95,8 +81,9 @@ describe("HttpConnection", () => {
 
                 await expect(connection.start(TransferFormat.Text))
                     .rejects
-                    .toThrow("Cannot start a connection that is not in the 'Disconnected' state.");
+                    .toThrow("Cannot start an HttpConnection that is not in the 'Disconnected' state.");
             } finally {
+                (options.transport as ITransport).onclose!();
```

Comment:
Before stop would complete before the transport closed. Now stop waits for the transport to close, so we indicate that the transport already closed before calling connection.stop() to ensure connection.stop() completes.

As for the as cast, I cant just do `options.transport!.onclose!();` because options.transport has the type `HttpTransportType | ITransport | undefined` instead of `ITransport | undefined`.

I could store the transport in a local, but meh.